metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -97,11 +106,30 @@ public FeatureControlManager build() {
                         MetadataVersion.latestProduction().featureLevel()));
                 quorumFeatures = new QuorumFeatures(0, localSupportedFeatures, List.of(0));
             }
+            if (kraftVersionAccessor == null) {
+                kraftVersionAccessor = new KRaftVersionAccessor() {
+                    private KRaftVersion version = KRaftVersion.LATEST_PRODUCTION;
```

Comment:
so if for some reason kraftVersionAccessor was not defined, we create a default accessor which says the current kraft version is the LATEST_PRODUCTION? any reason we don't just throw if the accessor is not assigned - is it just for ease of testing?

Reply:
This way avoids updating all the tests, which I think is a reasonable goal. If a test doesn't care about this it shouldn't have to set it.

Reply:
> any reason we don't just throw if the accessor is not assigned - is it just for ease of testing?

To make it easier to test this type. Builders in the metadata module tend to set the fields to a value that is valid for most tests.